.github/workflows/linting.yml

```diff
@@ -0,0 +1,22 @@
+name: Static Analysis
+on: [push, pull_request]
+jobs:
+  python-linting:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+
+    - name: install Python packages
+      run: |
+        pip install --upgrade pip
+        pip install --upgrade flake8
+
+    - name: Run flake8
+      run: |
+        flake8 easybuild/tools
```

Comment:
Can we also do a full run, but ignore any errors it produces, to assess how bad things are?

```shell
# run flake8 on full easybuild directory, but ignore the exit code until whole codebase is PEP8 compliant
flake8 easybuild || true

Reply:
I'd say, do this manually locally. This is a CI script so it shouldn't contain pointless stuff. I mean: What would you do with this information? Why do you need it for every CI run? You wouldn't even see it unless you look into the output and it may confuse users because it produces many messages and contributors are meant to look at this output to see what they need to fix to make a PR pass.

Reply:
Good point with the latter, so let's leave it as is.